### PR TITLE
Make Tensor.__contains__ accept SymInt/Float/Bool.

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98933

Fixes https://github.com/pytorch/pytorch/issues/98870

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

File: test/test_proxy_tensor.py [hunk 1]
```diff
@@ -1235,6 +1235,18 @@ def f(x, y):
...
 
+    def test_sym_contains(self):
+        def f(x, y):
+            return x.size(0) in y
+
+        # This shouldn't raise but we need SymBool from
+        # https://github.com/pytorch/pytorch/pull/98453
+        # then modify this.  It should NOT raise a RuntimeError
+        # though!
+        self.assertRaisesRegex(NotImplementedError, "item NYI for torch.bool", lambda:
```
Comment (albanD) [inline test/test_proxy_tensor.py:1246]: nit: you can use this as a context manager instead of passing a lambda if you need to break the call anyways.

**Review by albanD — APPROVED** [review]
SGTM

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**